src/core/objects/object.cpp

```diff
@@ -588,7 +588,7 @@ void Object::rotate(qreal angle)
 	{
 		auto const old_coord = MapCoordF(coord);
 		coord.setX(cos_angle * old_coord.x() + sin_angle * old_coord.y());
-		coord.setY(sin_angle * old_coord.x() + cos_angle * old_coord.y());
+		coord.setY(cos_angle * old_coord.y() - sin_angle * old_coord.x());
 	}
```

Comment:
I see, this hits the nail. This function is only used from TemplateTransform. However, for symmetry with `Obejct::rotateAround(const MapCoordF& center, qreal angle)` just above these lines, I would prefer to leae the x/y order and rather write:

~~~
		coord.setX(+ cos_angle * old_coord.x() + sin_angle * old_coord.y());
		coord.setY(- sin_angle * old_coord.x() - cos_angle * old_coord.y());
~~~

Reply:
Now that I've made the change, I see how it reflects the common layout of a rotation matrix.